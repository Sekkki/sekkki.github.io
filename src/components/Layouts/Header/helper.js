import React, { Component } from 'react';
import './helper.scss';

export default class Helper extends Component {

    constructor(props) {
        super(props);
        this.state = {
            status: false
        }
    }

    clickHandle() {
        this.setState({ status: !this.state.status });
    }

    render() {
        return (
            <div className="helper-wrapper"> 
                <div className="helper-container">
                    <div className="helper" onClick={() => this.clickHandle()}>
                    { !this.state.status ? <i className="fas fa-arrow-left"></i> : <i className="fas fa-arrow-right"></i> }
                    </div>
                    <div className={ !this.state.status ? "helper-button" : "helper-button is-open"}>
                        <i className="fas fa-sign-in-alt"></i>
                        <i className="fas fa-palette"></i>
                        <i className="fas fa-ruble-sign"></i>
                        <i className="fas fa-balance-scale"></i>
                    </div>
                </div>
            </div>    
        )
    }
}
