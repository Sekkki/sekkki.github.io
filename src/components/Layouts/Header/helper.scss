    .helper-container {
        border-radius: 10px;
        margin: 5px;
        position: fixed;
        z-index: 10;
        top: 0px;
        right: 0px;
        display: flex;
        box-shadow: 0px 0px 3px 0px #cdcdcd;

        .helper-button {
            background-color: rgb(228, 83, 83);
            display: flex;
            justify-content: center;
            margin-right: -180px;
            transition: margin 0.5s;
            word-wrap: break-all;

            i {
                overflow: hidden;
                color: rgb(114, 48, 48);
                font-size: 25px;
                vertical-align: middle;
                float: right;
                margin: 10px;

                &:hover {
                    transition: 0.2s;
                    color: #fff;
                //  color: rgb(168, 43, 43);
                }
            }
        }

        .helper-button.is-open {
            margin-right: 0px;
        }

        .helper {
            background-color: rgb(228, 83, 83);
            width: auto;
            display: flex;
            flex-direction: column;
            padding: 10px 10px 10px 15px;
            border-radius: 10px 0px 0px 10px;
            justify-content: center;
            background-color: rgb(228, 83, 83);
            text-align: center;
            i {
                transition: 0.5s;
                color: #fff;
                font-size: 20px;
                &:hover {
                    transform: rotate(360deg);
                }
            }
        }
    }

    @keyframes invise {

        100% {
            transform: scale(0.5);
            display: none;
        }
    }

    @media (max-width: 776px) {
        .helper-container {
            flex-direction: column;
            width: 50px;
            left: 0;
            .helper-button {
                margin-right: 0px;
                margin-top: -180px;
                display: flex;
                justify-content: center;
                flex-direction: column;
                i {
                    text-align: center;
                    float: left;
                }
            }
            .helper-button.is-open {
                margin-top: 0px !important;
            }

            .helper {
                order: 1;
                padding: 10px 0px;
                border-radius: 0px 0px 10px 10px !important;
                i {
                    transform: rotate(-90deg);
                    &:hover {
                        transform: rotate(270deg);
                    }
                }
            }
        }
    }