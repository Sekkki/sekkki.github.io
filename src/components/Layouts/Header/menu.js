import React, { Component } from 'react';
import './menu.scss';

export default class menu extends Component {

    render() {
        return (
            <div className={ this.props.show ? "mobile-menu-container show" : "mobile-menu-container"}> 
                <div className="mobile-menu">
                    <div className="mobile-menu-brand">Hyppnotic</div>
                    <div className="mobile-menu-button" onClick={() => this.props.setShow()}><i className="fas fa-times"></i></div>
                    <div className="mobile-menu-links">Links</div>
                </div>
            </div>
        )
    }
}
