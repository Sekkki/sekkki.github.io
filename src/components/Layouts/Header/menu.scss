@import '../../../variables.scss';

.mobile-menu-container {
    display: none;
    z-index: 100;
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    background-color: $main;
    .mobile-menu {
        display: block;
        width: 100%;
        height: 100%;
        position: relative;
        p {
            color: #fff;
        }
    }
    .mobile-menu-brand {

    }

    .mobile-menu-button {
        position: absolute;
        right: 0;
        top: 0;
        padding: 20px;
        i {
            font-size: 40px;
        }
        transition: .2s;
        &:hover {
            color: #fff;
        }
    }
}

.mobile-menu-container.show {
    display: block;
    overflow: hidden; 
}

@media (min-width: 577px) {
    .menu-container {
        display: none !important;
    }
}