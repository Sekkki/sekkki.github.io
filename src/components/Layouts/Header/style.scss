@import url('https://fonts.googleapis.com/css?family=Roboto:100,300,400,500&display=swap');

header {
    font-family: Roboto;
    .brand {
        padding: 50px 0px;
        p {
            text-align: center;
            letter-spacing: 0.2rem;
            font-size: 60px;
            font-weight: 300;
            color: #444040;
            span {
                font-size: 16px;
                letter-spacing: 0rem;
                font-weight: 400;
            }
        }
    }
    .menu {
        width: 100%;
        background-color: rgb(228, 83, 83);
        ul {
            width: 100%;
            padding: 0px;
            display: flex;
            justify-content: center;
            margin: 0px;
            li {
                font-size: 14px;
                padding: 25px 20px;
                font-weight: 500;
                list-style: none;
                color: rgb(255, 255, 255);
                transition: 0.3s;
                cursor: pointer;
                &:hover {
                    transform: scale(0.9);
                    background-color: rgb(255, 255, 255);
                    color: rgb(228, 83, 83);
                }
            }
        }
    }
    .button-xs {
        display: none;
        position: absolute;
        right: 0px;
        top: 0px;
        margin: 10px;
        width: 50px;
        height: 50px;
    }
}

@media (max-width: 576px) {
    header {
        position: sticky;
        top: 0;
        left: 0;
        width: 100%;
        z-index: 10;
        box-shadow: 0px 0px 3px 0px #cdcdcd;
        .brand {
            padding: 20px 0px;
            p {
                font-size: 20px;
                span {
                    font-size: 15px;
                }
            }
        }
        .menu {
            display: none;
        }
        .button-xs {
            display: flex;
            justify-content: center;
            i {
                color: rgb(228, 83, 83);
                font-size: 50px;
            }
        }
    }
}

@media (max-width: 776px) {
    header {
        .brand {
            padding: 20px 0px;
        }
    }
}