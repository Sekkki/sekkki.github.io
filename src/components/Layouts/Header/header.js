import React, { Component } from 'react'
import './style.scss';
import { Link } from 'react-router-dom';
import { Nav, Navbar, NavDropdown } from 'react-bootstrap';
import { withCookies } from 'react-cookie';
import Helper from './helper';
import Menu from './menu';
import ScrollReveal from 'scrollreveal';

class header extends Component {

    constructor(props) {
        super(props);
        this.state = {
            show: false
        }
    }

    componentDidMount() {
        if(this.state.show === true) {
            ScrollReveal().reveal('.show', { opacity: 0.8 });
            ScrollReveal().sync();
        }
    }

    show() {
        this.setState({ show: !this.state.show });
    }

    setShow() {
        this.setState({ show: false });
    }

    render() {

        const isAuthenticated = false;

        return (
            <header className="header">
                <div className="brand">
                    <p>Hyppnotic<span>store.</span></p>
                </div>
                <div className="menu">
                    <ul>
                        <li>Каталог</li>
                        <li>О нас</li>
                        <li>Помощь</li>
                        <li>Поддержка</li>
                    </ul>
                </div>
                <Helper />
                <Menu show={this.state.show} setShow={this.setShow.bind(this)} />
                <div className="button-xs" onClick={() => this.show()}><i className="fas fa-bars"></i></div>
            </header>
        )
    }
}

export default withCookies(header)
