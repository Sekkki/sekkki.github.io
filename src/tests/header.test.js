import React from 'react';
import Header from '../components/Layouts/Header/header';
import Helper from '../components/Layouts/Header/helper';
import Menu from '../components/Layouts/Header/menu';
import { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<Header />', () => {

it('renders without crashing', () => {
    jest.useFakeTimers();

    const wrapper = mount(<Header />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();

    jest.runAllTimers();

    expect(wrapper.find('.helper').simulate('click'));

    wrapper.unmount();
})

});

describe('<Helper />', () => {
    it('renders without crashing', () => {
        jest.useFakeTimers();

        const wrapper = mount(<Helper />);
    })
})