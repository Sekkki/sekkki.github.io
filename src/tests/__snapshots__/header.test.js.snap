// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders without crashing 1`] = `
<withCookies(header)>
  <withCookies(header)
    cookies={
      Cookies {
        "HAS_DOCUMENT_COOKIE": false,
        "changeListeners": Array [
          [Function],
        ],
        "cookies": Object {},
      }
    }
    forwardedRef={null}
  >
    <header
      allCookies={Object {}}
      cookies={
        Cookies {
          "HAS_DOCUMENT_COOKIE": false,
          "changeListeners": Array [
            [Function],
          ],
          "cookies": Object {},
        }
      }
    >
      <header
        className="header"
      >
        <div
          className="brand"
        >
          <p>
            Hyppnotic
            <span>
              store.
            </span>
          </p>
        </div>
        <div
          className="menu"
        >
          <ul>
            <li>
              Каталог
            </li>
            <li>
              О нас
            </li>
            <li>
              Помощь
            </li>
            <li>
              Поддержка
            </li>
          </ul>
        </div>
        <Helper>
          <div
            className="helper-wrapper"
          >
            <div
              className="helper-container"
            >
              <div
                className="helper"
                onClick={[Function]}
              >
                <i
                  className="fas fa-arrow-left"
                />
              </div>
              <div
                className="helper-button"
              >
                <i
                  className="fas fa-sign-in-alt"
                />
                <i
                  className="fas fa-palette"
                />
                <i
                  className="fas fa-ruble-sign"
                />
                <i
                  className="fas fa-balance-scale"
                />
              </div>
            </div>
          </div>
        </Helper>
        <menu
          setShow={[Function]}
          show={false}
        >
          <div
            className="mobile-menu-container"
          >
            <div
              className="mobile-menu"
            >
              <div
                className="mobile-menu-brand"
              >
                Hyppnotic
              </div>
              <div
                className="mobile-menu-button"
                onClick={[Function]}
              >
                <i
                  className="fas fa-times"
                />
              </div>
              <div
                className="mobile-menu-links"
              >
                Links
              </div>
            </div>
          </div>
        </menu>
        <div
          className="button-xs"
          onClick={[Function]}
        >
          <i
            className="fas fa-bars"
          />
        </div>
      </header>
    </header>
  </withCookies(header)>
</withCookies(header)>
`;
